app/controllers/quiz_controller.rb

```diff
@@ -27,14 +28,12 @@ def task
         end
       end
     when 6
-      answer = $data_67[params["question"].del_punc.chars.sort]
+      answer = $data_67[params["question"].del_punc.chars.sort].gsub(" —","")
     when 7
-      answer = $data_67[params["question"].del_punc.chars.sort]
+      answer = $data_67[params["question"].del_punc.chars.sort].gsub(" —","")
     when 8
-      # mas = params["question"].del_punc.chars.sort
-      # mas.each_with_index do |t, i|
-      #   break if (answer = $data_8[[mas[0...i], mas[(i+1)..-1]].flatten])
-      # end
+      mas = params["question"].del_punc.chars
+      answer = $data_8_[mas.size].select{|k,v| (1..2).include? (mas - k | k - mas).size}.first.last.gsub(" —","")
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.